actioncable/lib/action_cable/channel/base.rb

```diff
@@ -153,6 +155,14 @@ def initialize(connection, identifier, params = {})
         subscribe_to_channel
       end
 
+      # This method is called after subscription has been added to the channel.
+      # Send confirmation here to avoid race conditions when client tries to perform actions
+      # right after receiving confirmation.
+      def registered!
```

Comment:
`registered` also adds an extra concept that closely mirrors `subscribed` and `confirmed` — is it meant to communicate that a subscription is about to be confirmed? Could we make that intent come out further?